server/store/datastore/xorm.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datastore
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+	xlog "xorm.io/xorm/log"
+)
+
+func newXORMLogger(level xlog.LogLevel) xlog.Logger {
+	return &xormLogger{
+		level: level,
+	}
+}
+
+// xormLogger custom log implementation for ILogger
+type xormLogger struct {
+	level   xlog.LogLevel
+	showSQL bool
+}
+
+// Error implement ILogger
+func (x *xormLogger) Error(v ...interface{}) {
+	if x.level <= xlog.LOG_ERR {
+		log.Error().Msg(fmt.Sprintln(v...))
```

Comment:
we should add some string to indicate it's from xorm

log.Str("component", "xorm")

or log.Bool("xorm": true)
....

?

Reply:
what do you think?

Reply:
Yeah, that makes sense to me :+1:  Will finish the PR later.

Reply:
I have used the `"component", "xorm"` format. If someone wants to parse the logs in a central log mgm system, using a consistent field to get the component name might be better than a changing bool field.